AWS_Backup/AMI_Backup.py

```diff
@@ -0,0 +1,245 @@
+
+#! /usr/bin/python
+
+import argparse
+import datetime
+import json
+import pytz
+import requests
+import boto3
+
+Utc = pytz.UTC
+
+def slack(event, channel, webhookurl):
+    #This function is used to post message to slack
+    message = 'AMI_BKP_ALERT'
+    payload = {'channel': channel,
+               'username': "AMI_BKP_SCRIPT",
+               'text': message,
+               'attachments': [{
+                   'color': 'danger',
+                   'fields': [{'value': event}]
+                   }]
+              }
+    req = requests.post(webhookurl, data=json.dumps(payload))
+
+def amibkp(region, days_del, slack_req, slack_channel, slack_webhook):
+    """
+    This function is the crucial function,
+    fetches all the instances which has tag Key:AMIBACKUPON, Value:yes and creates AMI in a loop,
+    along with propogating all the tags from instance to AMI to EBS Snapshots.
+    Also, it deletes all the AMI's which was created through this script
+    and older than number of days you provide as an argument.
+    Parameters
+    ----------
+    region: string
+          AWS region code.
+    days: integer
+          Number of days to keep AMI's before deleting.
+    slack: string
+          Optional argument.
+          Passing this parameter as "true" will post the execption to slack if any.
+    slack_channel: String
+          Slack channel to where exceptions has to be posted
+          Depends on the previous parameter "slack", this is required if slack is true.
+    webhookurl: string
+          Slack webhookurl to identify to which slack team exeception has to be posted?
+          Depends on the previous parameter "slack", this is required if slack is true.
+    Returns
+    -------
+    list
+        Returns list of AMI's/Snapshots deleted and newly created AMI's/Snapshots.
+    """
+    client = boto3.client('ec2', region_name=region)
+    ec2 = boto3.resource('ec2', region_name=region)
+    response2 = client.describe_images(
```

Comment:
Can we name this better, something like `describe_images_response`?

Reply:
I have made the changes